Feature/auto publish updates

pverscha opened the pull request:
This PR introduces a new GitHub action that is automatically run every 2 months. This action generates new versions of the underlying data for our application (derived from SwissProt releases) and automatically publishes a new version of the application on PyPi.

rababerladuseladim commented on .github/workflows/publish.yml:
```diff
@@ -0,0 +1,49 @@
+        env:
+          TWINE_USERNAME: ${{ secrets.TWINE_USERNAME }}
+          TWINE_PASSWORD: ${{ secrets.TWINE_PASSWORD }}
+
```
Awesome! I will hopefully remember this action if I ever want to do something similar.

rababerladuseladim replied:
One more remark: how about restricting the publishing to the master branch? Simply to avoid publishing feature/bugfix-branches by accident.

pverscha replied:
I would rather disable automatic publishing all together (except for the automatic bi-monthly release). That way we can decide to bundle multiple PR's and release them at once

rababerladuseladim replied:
I absolutely agree, but right now it's possible to trigger the action on branches, isn't it?

pverscha replied:
Yes, it is :) This was merely something I added to test the publishing feature, but I'll remove it together with the other changes I need to perform to this PR :) Should be done by tonight! I'll re-request your review once this has been done

rababerladuseladim commented on .github/workflows/tests.yml:
```diff
@@ -1,21 +1,30 @@
-      - name: Install required dependencies
+      - name: Override database files (these need to be fixed since the test values depend on the version of the dataset used)
+        run: rm -rf megago/resources && mv functional_tests/resources/ megago/resources/
+      - name: Install required dependencies on ${{ matrix.os }}
+        if: matrix.os != 'windows-latest'
+        run: |
+          python3 -m pip install --upgrade pip
+          pip3 install -r requirements-dev.txt
+          pip3 install -U .
+      - name: Install required dependencies on ${{ matrix.os }}
+        if: matrix.os == 'windows-latest'
         run: |
           python -m pip install --upgrade pip
           pip install -r requirements-dev.txt
```
Windows is not tested anymore, is it? Besides, this looks like a duplicate: step `Install required dependencies on ${{ matrix.os }}` exists twice.

pverscha replied:
I still want to test Windows, I just need to split up the test workflows cause there are too many differences between running it on a *nix-based OS and Windows. I'll make sure to include that before I re-request a review from you :) Thanks for pointing this out. Both steps with the if are actually different (one uses `python3` and `pip3` commands, the other one `python` and `pip` as required by Windows)

rababerladuseladim reviewed (changes requested):
Great work, I do have som equestions though: 
1) is windows support ditched altogether? If it is, windows specific code should be removed from the actions entirely.
2) What do you think about logging the data version? This is mandatory for reproducibility, in my opinion.

pverscha commented:
@rababerladuseladim We still support Windows, I forgot to update the Windows workflow. I'll indeed add the version number to the log. Thanks for your feedback :)

pverscha commented:
Hi @rababerladuseladim,

I fixed the tests on Windows, I added the version number to the log file of the application and the auto publisher is now configured in such a way that it does not react to PR's or pushes. Could you take another look? :)

rababerladuseladim commented on .github/workflows/publish.yml:
```diff
@@ -0,0 +1,51 @@
+on:
+  schedule:
+    - cron: '0 0 1 */2 *'
+  workflow_dispatch:
```
final question: shouldn't the `on` section include a restriction to the master branch, especially since the publishing is triggered automatically.
